cli/script/command-executor.ts

```diff
@@ -438,9 +449,24 @@ function initiateExternalAuthenticationAsync(serverUrl: string, action: string):
 }
 
 function login(command: cli.ILoginCommand): Promise<void> {
-    initiateExternalAuthenticationAsync(command.serverUrl, "login");
+    // Check if one of the flags were provided.
+    if (command.accessKeyName || command.providerName || command.providerUniqueId) {
+        throwForMissingCredentials(command.accessKeyName, command.providerName, command.providerUniqueId);
+        
+        sdk = new AccountManager(command.serverUrl);
+        var accessToken: string = base64.encode(JSON.stringify({ accessKeyName: command.accessKeyName, providerName: command.providerName.toLowerCase(), providerUniqueId: command.providerUniqueId }));
```

Comment:
not super important, but, create object on a separate line and encode it here. Makes it easy for readability and debugging, just for future reference.